src/formats/pam.zig

```diff
@@ -0,0 +1,492 @@
+const std = @import("std");
+const io = std.io;
+const mem = std.mem;
+const math = std.math;
+const ascii = std.ascii;
+const fmt = std.fmt;
+const meta = std.meta;
+const Allocator = std.mem.Allocator;
+const color = @import("../color.zig");
+const FormatInterface = @import("../format_interface.zig").FormatInterface;
+const PixelStorage = color.PixelStorage;
+const PixelFormat = @import("../pixel_format.zig").PixelFormat;
+const Image = @import("../Image.zig");
+const ImageError = Image.Error;
+const ImageReadError = Image.ReadError;
+const ImageWriteError = Image.WriteError;
+const utils = @import("../utils.zig");
+
+const TupleType = enum {
+    mono,
+    mono_a,
+    gray,
+    gray_a,
+    rgb,
+    rgb_a,
+
+    fn fromString(s: []const u8) error{Unsupported}!TupleType {
```

Comment:
I chose to use `error{Unsupported}` to clarify that `Unsupported` is the only error that can be returned from `fromString`, as the other errors (e.g. `error.InvalidData`) do not make sense in this context and `TupleType` is entirely disconnected from Image, using `ImageError` would create coupling where it does not make sense to have it.

Reply:
We want to avoid using unscoped errors to control the scope of errors we expect the user to handle. In this case, `ImageError` only contains `Unsupported` in this case.